sdk-exporter/prometheus/src/main/scala/org/typelevel/otel4s/sdk/exporter/prometheus/PrometheusMetricExporter.scala

```diff
@@ -0,0 +1,251 @@
+    def withWithoutUnits(withoutUnits: Boolean): HttpServerBuilder[F]
+
+    /** Sets the flag indicating if metrics are produced without a type suffix. */
+    def withWithoutTypeSuffixes(withoutTypeSuffixes: Boolean): HttpServerBuilder[F]
+
+    /** Sets the flag indicating if metrics are produced without a scope info metric or scope labels. */
+    def withDisableScopeInfo(disableScopeInfo: Boolean): HttpServerBuilder[F]
+
+    /** Sets the flag indicating if metrics are produced without a target info metric. */
+    def withDisableTargetInfo(disableTargetInfo: Boolean): HttpServerBuilder[F]
+
+    /** Creates a [[PrometheusMetricExporter]] using the configuration of this builder and starts an HTTP server that
+      * will collect metrics and serialize to Prometheus text format on request.
+      */
+    def build: Resource[F, MetricExporter.Pull[F]]
+  }
+
+  /** Creates a [[Builder]] for [[PrometheusMetricExporter]] with the default configuration.
+    */
+  def builder[F[_]: Concurrent: Console]: Builder[F] =
+    new BuilderImpl[F](
+      defaultAggregationSelector = AggregationSelector.default
+    )
+
+  /** Creates a [[HttpServerBuilder]] for [[PrometheusMetricExporter]] with the default configuration.
+    */
+  def serverBuilder[F[_]: Async: Network: Console]: HttpServerBuilder[F] =
+    new HttpServerBuilderImpl[F](
+      host = Defaults.Host,
+      port = Defaults.Port,
+      defaultAggregationSelector = AggregationSelector.default,
+      withoutUnits = Defaults.WithoutUnits,
+      withoutTypeSuffixes = Defaults.WithoutTypeSuffixes,
+      disableScopeInfo = Defaults.DisableScopeInfo,
+      disableTargetInfo = Defaults.DisableTargetInfo
+    )
+
+  private final case class BuilderImpl[F[_]: Concurrent: Console](
+      defaultAggregationSelector: AggregationSelector
+  ) extends Builder[F] {
+    def withDefaultAggregationSelector(
+        selector: AggregationSelector
+    ): Builder[F] = copy(defaultAggregationSelector = selector)
+
+    def build: F[MetricExporter.Pull[F]] =
+      Ref.empty[F, Vector[MetricProducer[F]]].map { ref =>
+        new PrometheusMetricExporter[F](
+          ref,
+          defaultAggregationSelector
+        )
+      }
+
+  }
+
+  private final case class HttpServerBuilderImpl[F[_]: Async: Network: Console](
+      host: Host,
+      port: Port,
+      defaultAggregationSelector: AggregationSelector,
+      withoutUnits: Boolean,
+      withoutTypeSuffixes: Boolean,
+      disableScopeInfo: Boolean,
+      disableTargetInfo: Boolean
+  ) extends HttpServerBuilder[F] {
+    def withHost(host: Host): HttpServerBuilder[F] =
+      copy(host = host)
+
+    def withPort(port: Port): HttpServerBuilder[F] =
+      copy(port = port)
+
+    def withDefaultAggregationSelector(
+        selector: AggregationSelector
+    ): HttpServerBuilder[F] = copy(defaultAggregationSelector = selector)
+
+    def withWithoutUnits(withoutUnits: Boolean): HttpServerBuilder[F] =
```

Comment:
I have mixed feelings about `withWithoutX`. But it's nearly identical to OtelJava API, which may be a plus (is it?).

Also, if we move forward with the `PrometheusWriter.Config`, we can use it here :
```scala
def withWriterConfig(config: PrometheusWriter.Config): HttpServerBuilder[F] = ... 
```

Reply:
`withWriterConfig` is the option I ended up choosing to use, as I mentioned in the comment above.